src/main/java/com/github/osisoft/ocs_sample_library_preview/BaseClient.java

```diff
@@ -112,21 +124,34 @@ public BaseClient(String apiVersion, String tenantId, String clientId, String cl
      * @return
      */
     public HttpURLConnection getConnection(URL url, String method) {
+        return getHttpURLConnection(url, method, Collections.<String, String>emptyMap());
+    }
+
+    private HttpURLConnection getHttpURLConnection(URL url, String method, Map<String, String> headers) {
         HttpURLConnection urlConnection = null;
         String token = "";
         if (!this.ClientId.isEmpty()) {
             token = AcquireAuthToken();
         }
-
+        
         try {
             urlConnection = (HttpURLConnection) url.openConnection();
             urlConnection.setRequestMethod(method);
-            urlConnection.setRequestProperty("Accept", "*/*; q=1");
-            urlConnection.setRequestProperty("Accept-Encoding", "gzip");
-            urlConnection.setRequestProperty("Content-Type", "application/json");
+            
+            if (headers.equals(Collections.<String, String>emptyMap())) {
```

Comment:
Not necessary see comment above. If you want the ability for an emptyMap to be handled, I would recommend setting headers equal to getHttpHeadersForRequest() instead.